test/project_types/script/layers/domain/extension_point_test.rb

```diff
@@ -15,15 +15,29 @@
   end
 
   describe ".new" do
-    subject { Script::Layers::Domain::ExtensionPoint.new(type, config) }
-    it "should construct new ExtensionPoint" do
-      extension_point = subject
-      assert_equal type, extension_point.type
+    describe "when deprecation status is not specified" do
+      subject { Script::Layers::Domain::ExtensionPoint.new(type, config) }
+      it "should construct new, non-deprecated ExtensionPoint" do
+        extension_point = subject
+        assert_equal type, extension_point.type
+        refute extension_point.deprecated?
 
-      sdk = extension_point.sdks[:ts]
-      refute_nil sdk.package
-      refute_nil sdk.sdk_version
-      refute_nil sdk.toolchain_version
+        sdk = extension_point.sdks[:ts]
+        refute_nil sdk.package
+        refute_nil sdk.sdk_version
+        refute_nil sdk.toolchain_version
+      end
+
+      describe "when deprecation status is specified" do
+        let(:config_with_deprecation) { config.merge({ "deprecation" => true }) }
+
+        subject { Script::Layers::Domain::ExtensionPoint.new(type, config) }
+
+        it "should construct a deprecated extension point" do
+          extension_point = subject
+          refute extension_point.deprecated?
```

Comment:
I think this specific test should `assert extension_point.deprecated?`, right?

Reply:
Ah right, whoops